make/autoconf/jdk-options.m4

```diff
@@ -438,6 +438,14 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
           # It's harmless to be suppressed in clang as well.
           ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
           ASAN_LDFLAGS="-fsanitize=address"
+          # detect_stack_use_after_return is using a fake stack and thus JDK would crashe:
+          # assert(_thread->is_in_live_stack((address)this)) failed: not on stack?
+          if test "x$TOOLCHAIN_TYPE" = "xgcc"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS --param asan-use-after-return=0"
+          fi
+          if test "x$TOOLCHAIN_TYPE" = "xclang"; then
+            ASAN_CFLAGS="$ASAN_CFLAGS -fsanitize-address-use-after-return=never"
+          fi
```

Comment:
I don't think this change should be made.  Globally disabling some otherwise
likely useful asan warnings to address what looks like a localized problem
doesn't seem like a good approach to me.

Also, suppression seems inappropriate since this has the look of a possible
JVM bug. It's hard to tell without more context for the failure, like the full
stack trace and how to reproduce.

Reply:
I understand the problem. You are right there are multiple (2?) ways how to fix it. A reproducer simulating what JDK does:
```
#include <cstdio>
struct C { C() {
    printf("C this=%p sp=%p\n",this,__builtin_frame_address(0));
    puts(this>=__builtin_frame_address(0)?"PASS":"FAIL");
} };
int main() { C c;
  puts("exit"); }
```
The result is:
```
$ gcc -o asan asan.C -Wall -g -fsanitize=address;ASAN_OPTIONS=detect_stack_use_after_return=0 ./asan
C this=0x7fffd9dfb720 sp=0x7fffd9dfb6f0
PASS
$ gcc -o asan asan.C -Wall -g -fsanitize=address;ASAN_OPTIONS=detect_stack_use_after_return=1 ./asan
C this=0x7f08f6009020 sp=0x7ffef58196a0
FAIL
```
The compiler options in this patch just moves the `ASAN_OPTIONS=detect_stack_use_after_return=X` runtime configuration to compile time.

https://github.com/google/sanitizers/wiki/AddressSanitizerUseAfterReturn

The other option is to disable the `is_in_live_stack` assertion in the case of `#ifdef ADDRESS_SANITIZER` (or even detect somehow whether the ASAN option `detect_stack_use_after_return` is active, I did not investigate that).